cpp/src/stream_compaction/filter/filter.cu

```diff
@@ -228,17 +228,18 @@ jitify2::ConfiguredKernel build_kernel(std::string const& kernel_name,
                                        std::vector<column_view> const& input_columns,
                                        bool has_user_data,
                                        std::string const& udf,
-                                       bool is_ptx,
+                                       udf_source_type source_type,
                                        rmm::cuda_stream_view stream,
                                        rmm::device_async_resource_ref mr)
 {
   auto const cuda_source =
-    is_ptx ? cudf::jit::parse_single_function_ptx(
-               udf,
-               "GENERIC_FILTER_OP",
-               cudf::jit::build_ptx_params(
-                 span_outputs, cudf::jit::column_type_names(input_columns), has_user_data))
-           : cudf::jit::parse_single_function_cuda(udf, "GENERIC_FILTER_OP");
+    (source_type == udf_source_type::PTX)
+      ? cudf::jit::parse_single_function_ptx(
+          udf,
+          "GENERIC_FILTER_OP",
+          cudf::jit::build_ptx_params(
+            span_outputs, cudf::jit::column_type_names(input_columns), has_user_data))
+      : cudf::jit::parse_single_function_cuda(udf, "GENERIC_FILTER_OP");
```

Comment:
nit: Maybe more readable if we use `if-else` instead of ternary?

Reply:
Agreed.